src/EnergyPlus/GroundHeatExchangers.cc

```diff
@@ -163,52 +156,829 @@ namespace GroundHeatExchangers {
+
+		} else {
+			
+			z = Z0 + std::sin( theta ) * ( coilDiameter / 2.0 );
+
+			zIn = Z0 + std::sin( eta ) * ( coilDiameter / 2.0 - pipeOuterRadius );
+			zOut = Z0 + std::sin( eta ) * ( coilDiameter / 2.0 + pipeOuterRadius );
+
+			return 0.5 * std::sqrt( std::pow( x - xIn, 2.0 ) + std::pow( Y0( m1 ) - Y0( m ), 2.0 ) + std::pow( z - zIn, 2.0 ) ) 
+				+ 0.5 * std::sqrt( std::pow( x - xOut, 2.0 ) + std::pow( Y0( m1 ) - Y0( m ), 2.0 ) + std::pow( z - zOut, 2.0 ) );			
+		}
+	};
+
+	//******************************************************************************
+
+	Real64
+	GLHESlinky::distanceToFictRing(
+		int const m,
+		int const n,
+		int const m1,
+		int const n1,
+		Real64 const eta,
+		Real64 const theta	
+	)
+	{
+
+		// SUBROUTINE INFORMATION:
+		//       AUTHOR:          Matt Mitchell
+		//       DATE WRITTEN:    February, 2015
+		//       MODIFIED         na
+		//       RE-ENGINEERED    na
+
+		// PURPOSE OF THIS SUBROUTINE:
+		// Calculates the distance between any two points
+
+		// METHODOLOGY EMPLOYED:
+
+		// REFERENCES:
+
+		// USE STATEMENTS:
+
+		// SUBROUTINE ARGUMENT DEFINITIONS:
+
+		// SUBROUTINE PARAMETER DEFINITIONS:
+
+		// INTERFACE BLOCK SPECIFICATIONS:
+
+		// DERIVED TYPE DEFINITIONS:
+
+		// SUBROUTINE LOCAL VARIABLE DECLARATIONS:
+
+		// Function to calculate distance between any two point on any two loops
+
+		Real64 x;
+		Real64 y;
+		Real64 z;
+		Real64 xIn;
+		Real64 yIn;
+		Real64 zIn;
+		Real64 xOut;
+		Real64 yOut;
+		Real64 zOut;
+		Real64 pipeOuterRadius;
+
+		pipeOuterRadius = pipeOutDia / 2.0;
+
+		x = X0( n ) + std::cos( theta ) * ( coilDiameter / 2.0 );
+		y = Y0( m ) + std::sin( theta ) * ( coilDiameter / 2.0 );
+		z = Z0 + std::sin( theta ) * ( coilDiameter / 2.0 ) + 2 * coilDepth;
+
+		xIn = X0( n1 ) + std::cos( eta ) * ( coilDiameter / 2.0 - pipeOuterRadius );
+		yIn = Y0( m1 ) + std::sin( eta ) * ( coilDiameter / 2.0 - pipeOuterRadius );
+		zIn = Z0 + std::sin( eta ) * ( coilDiameter / 2.0 - pipeOuterRadius );
+
+		xOut = X0( n1 ) + std::cos( eta ) * ( coilDiameter / 2.0 + pipeOuterRadius );
+		yOut = Y0( m1 ) + std::sin( eta ) * ( coilDiameter / 2.0 + pipeOuterRadius );
+		zOut = Z0 + std::sin( eta ) * ( coilDiameter / 2.0 + pipeOuterRadius );
+
+		return 0.5 * std::sqrt( std::pow( x - xIn, 2.0 ) + std::pow( Y0( m1 ) - Y0( m ), 2.0 ) + std::pow( z - zIn, 2.0 ) ) 
+				+ 0.5 * std::sqrt( std::pow( x - xOut, 2.0 ) + std::pow( Y0( m1 ) - Y0( m ), 2.0 ) + std::pow( z - zOut, 2.0 ) );			
+
+	};
+	
+	//******************************************************************************
+
+	Real64
+	GLHESlinky::distToCenter(
+		int const m, 
+		int const n,
+		int const m1,
+		int const n1
+	)
+	{
+
+		// SUBROUTINE INFORMATION:
+		//       AUTHOR:          Matt Mitchell
+		//       DATE WRITTEN:    February, 2015
+		//       MODIFIED         na
+		//       RE-ENGINEERED    na
+
+		// PURPOSE OF THIS SUBROUTINE:
+		// Calculates the center-to-center distance between rings
+
+		// METHODOLOGY EMPLOYED:
```

Comment:
Another random note: I'm less concerned about all these stubs.  I like having some information in the top of the function to help get author/written/methodology/purpose/references where applicable.  But feel free to clean out the bulk of these that aren't used.